app/middlewares/schema-validation.js

```diff
@@ -0,0 +1,24 @@
+const { body, validationResult, checkSchema } = require('express-validator');
+const { User } = require('../../app/models');
+const { EMAIL_ALREADY_EXISTS } = require('../errors');
+const { validationError, conflictError } = require('../errors');
+
+exports.schemaValidation = schema => [
+  checkSchema(schema),
+  body('email').custom(async value => {
```

Comment:
You can do custom validator on the schema or do a separate middleware for this

Reply:
Done